components/brave_news/browser/feed_generation_info.cc

```diff
@@ -285,12 +285,10 @@ mojom::FeedItemMetadataPtr FeedGenerationInfo::PickAndConsume(
   }
 
   auto index = maybe_index.value();
-  if (index >= articles.size()) {
-    DCHECK(false) << "|index| should never be outside the bounds of |articles| "
-                     "(index: "
-                  << index << ", articles.size(): " << articles.size();
-    return nullptr;
-  }
+  CHECK_LT(index, articles.size())
+      << "|index| should never be outside the bounds of |articles| "
+         "(index: "
+      << index << ", articles.size(): " << articles.size();
```

Comment:
nit: I don't think this detailed explanation is needed here. `CHECK_LT(index, articles.size());` should be enough.

Reply:
Fixed.